### ISSUE TITLE
Index names collide

### ISSUE THREAD

**Issue by cgardens** [airbytehq/airbyte#2948]
## Expected Behavior
* If I have very long and similar table names, I should be able to to replicate them into postgres.

## Current Behavior
* If I have very long and similar table names, they fail when replicating into postgres because the truncated index / constraint names collide.
* This is a big surprising as I thought these constraints would be scoped by table.

## Logs
from slack convo: https://airbytehq.slack.com/archives/C01MFR03D5W/p1618845804115300?thread_ts=1618588773.093500&cid=C01MFR03D5W
```
ERROR: duplicate key value violates unique constraint "_airbyte_raw_repairpal_production_carmax_claim_customer_re_pkey"
Detail: Key (_airbyte_ab_id)=(fee7c268-6423-44c0-8d5d-31a42a484983) already exists.
```

```
running SELECT * FROM  <table name> WHERE _airbyte_ab_id ~ 'fee7c268*'
```
Only returns one object
```
SELECT (<table name>)::text, count(_airbyte_ab_id)
FROM <table name>
GROUP BY <table name>._airbyte_ab_id
HAVING count(_airbyte_ab_id) > 1 
```
also doesn’t return anything

but was able to find the cause
I have two tables with long and similar names and I also have a long ingest db name in postgres
so I can sync one table at a time but if I sync both
_airbyte_\d{13}_<db>_<table>_key <- this key is only 64 characters long and it gets set from the table before. (

## Airbyte Version
0.20.0

## Connector Version (if applicable)
Postgres 0.24.0

## Additional context
The source being pulled from is mysql 0.25, though that should not be relevant to the bug fix.

**Comment by cgardens** [comment]
blocked on update to configured catalog.

**Comment by grishick** [comment]
@cgardens is this still blocked?

**Comment by etsybaev** [comment]
Basically, we already had a similar issue and had a follow-up ticket (https://github.com/airbytehq/airbyte/issues/13262).
Unfortunately, there is no way to fix it on a connector's level.
For more details please also check this Proposal:

https://docs.google.com/document/d/1UqLrnISRd3yWv5wiE5Z5QhU3L1KMpq3YQvAi9HRbcwk/edit
